dds/DCPS/ReactorTask.cpp

```diff
@@ -272,6 +323,42 @@ String ThreadStatus::get_key(const char* safety_profile_tid, const String& name)
   return key;
 }
 
+bool ThreadStatusManager::sync_with_parent(ThreadStatusManager& parent,
+  ThreadStatusManager::Map& running, ThreadStatusManager::Map& finished)
+{
+  ACE_WRITE_GUARD_RETURN(ACE_Thread_Mutex, g1, lock_, false);
+
+  {
+    ACE_READ_GUARD_RETURN(ACE_Thread_Mutex, g2, parent.lock_, false);
```

Comment:
I know this is not new to this PR, but with `ACE_Thread_Mutex` there is no use in distinguishing between `READ` and `WRITE` mode locks.  I am not advocating for changing to a more complex sync object.